geoh5py/data/primitive_type_enum.py

```diff
@@ -39,3 +43,23 @@ class PrimitiveTypeEnum(Enum):
     GEOMETRIC = 9
     MULTI_TEXT = 10
     BOOLEAN = 11
+
+
+def convert_to_primitive_type(data: np.ndarray, primitive_type: str) -> np.ndarray:
```

Comment:
I like the idea a lot. See comment above so we can leverage this in the values setter of NumericData.

Reply:
I changed convert_to_primitive_type in the numeric Datatype.

In order to respect the type of the data. 


New "check_type" function. 
Float can take any number,
Integer can take number without decimal.
Booleans can just take 0,1 (or TRUE, FALSE).

function ckeck_vector_length change to "format_value" in order to do everything.